com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -48,6 +48,7 @@ enum ShaderFeatures
         MainLightShadowsCascade = (1 << 26),
         DrawProcedural = (1 << 27),
         ScreenSpaceOcclusionAfterOpaque = (1 << 28),
+        ShadowsKeepOfVariant = (1 << 29),
```

Comment:
```suggestion
        ShadowsKeepOffVariant = (1 << 29),
```

Reply:
We need to start thinking about that we're quickly running out of bits here...

Reply:
This is for shadow and additional light keywords. It needs somekind of prefix to indicate that it does not control for all features

Reply:
Yeap this needs to be struct in my opnion, but for time being I think it can be 64bit too

Reply:
Agree, we just felt like bringing the bit issue up, but not that it should be fixed in this PR. And yeah, I agree with struct being a good choice instead.

Regarding the name, could it be something like `LightingKeepOffVariants`? IMO that's covers shadows as well. Maybe with a comment somewhere about what it exactly covers.

Reply:
or maybe just go all in and call it `KeepOffVariantsShadowsAdditionalLights`, although that is a bit long 😅